link/link_test.go

```diff
@@ -91,6 +91,16 @@ func mustCgroupFixtures(t *testing.T) (*os.File, *ebpf.Program) {
 	testutils.SkipIfNotSupported(t, haveProgAttach())
 
 	prog := mustCgroupEgressProgram(t)
+	cgroup := mustCgroupFixturesWithProgram(t, prog)
+
+	return cgroup, prog
+}
+
+func mustCgroupFixturesWithProgram(t *testing.T, prog *ebpf.Program) *os.File {
```

Comment:
I kind of expected this function to attach the program to the cgroup, but that doesn't seem to happen? Why pass program here?

Reply:
I think you can move this function into `internal/testutils/cgroup.go` and then keep `TestCgroupPerCPUStorageMarshaling` in `map_test.go`. That resolves your import cycle while keeping the test closer to the other per cpu tests.

Reply:
I tried to follow your suggestion, but ended up with another go compiler import cycle. I'm not sure what to do / if this is worth the trouble. I'll push the (broken) WIP in a separate commit.